src/conn/conn_ckpt.c

```diff
@@ -38,6 +38,13 @@ __ckpt_server_config(WT_SESSION_IMPL *session, const char **cfg, bool *startp)
 	if (conn->ckpt_usecs != 0 ||
 	    (conn->ckpt_logsize != 0 &&
 	    FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED))) {
+		/*
+		 * If non-zero, use a minimum of the log file size.  The logging
+		 * subsystem has already been initialized.
+		 */
+		if (conn->ckpt_logsize != 0)
+			conn->ckpt_logsize = WT_MAX(
+			    conn->ckpt_logsize, conn->log_file_max);
```

Comment:
I bolstered the conditional to check the flag if logging is enabled.  If it is, we are sure that log_file_max is set.  I added some test cases to one of the reconfig unit tests.  I checked that it did the right thing via a printf.